pulsar-core/src/event.rs

```diff
@@ -164,25 +165,33 @@ pub enum Payload {
         len: usize,
         is_tcp: bool,
     },
-    MalwareDetection {
-        score: f32,
-        #[validatron(skip)]
-        tags: Vec<String>,
-    },
-    RuleEngineDetection {
-        #[validatron(skip)]
-        rule_name: String,
-        #[validatron(skip)]
-        payload: Box<Payload>,
-    },
-    AnomalyDetection {
-        score: f32,
-    },
+    Empty,
+
+    // MalwareDetection {
+    //     score: f32,
+    //     #[validatron(skip)]
+    //     tags: Vec<String>,
+    // },
+    // RuleEngineDetection {
+    //     #[validatron(skip)]
+    //     rule_name: String,
+    //     #[validatron(skip)]
+    //     payload: Box<Payload>,
+    // },
+    // AnomalyDetection {
+    //     score: f32,
+    // },
     // CustomJson { ty: i32, data: Vec<u8> },
     // CustomProto { ty: i32, data: Vec<u8> },
     // CustomRaw { ty: i32, data: Vec<u8> }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize,)]
+pub struct Threat {
+    pub source: ModuleName,
+    pub info: HashMap<String, String>
```

Comment:
Not sure about this. Compared a `serde_json::Value`:
PRO: easier to print to stdout
CON: can't use serde for Serialize/Deserialize custom struct
CON: no hierarchical data structures and lists